cpp-package/include/mxnet-cpp/.gitignore

```diff
@@ -0,0 +1,2 @@
+# Rebuildable file(s)
+op.h
```

Comment:
Why is op.h placed inside the build dir? Usually we only support out of source build in cmake

Reply:
op.h is generated by OpWrapperGenerator.py from user side, which should not be tracked by git. Should I just put this .gitignore in cpp-package directory or merged with incubator-mxnet/.gitignore

Reply:
> op.h is generated by OpWrapperGenerator.py from user side

OpWrapperGenerator.py is invoked by cmake inside the `build` folder ("out of source build"). Usually out-of-source build should not modify the code / files outside of the `build` folder.